### PR TITLE
Update release notes

### PR DESCRIPTION
**Pull request by jarrodmillman**
The list of commits and authors was updated via:
```
# checkout branches and make sure they are up-to-date
git checkout networkx-2.8.8
git pull origin networkx-2.8.8
git checkout main
git pull origin main

# find out what commits on main haven't been cherry-picked already
git cherry -v networkx-2.8.8 main > cherry-log.txt

# get their commit messages and hashes
grep "^+" cherry-log.txt | cut -d " " -f 3- > 3.0-new-commit-messages.txt
grep "^+" cherry-log.txt | cut -d " " -f 2  > 3.0-new-commit-hashes.txt

# get their commit authors
> 3.0-new-commit-authors.txt
lines=$(cat 3.0-new-commit-hashes.txt)
for line in $lines
do
   git log --format="%an" ${line}^! >> 3.0-new-commit-authors.txt
done

# get a unique sorted list of authors
sort -u -o 3.0-new-commit-authors.txt{,}

# make messages and authors into markdown lists
awk -i inplace '{ print "-", $0 }' 3.0-new-commit-messages.txt
awk -i inplace '{ print "-", $0 }' 3.0-new-commit-authors.txt
```

### CODE REVIEW

**Comment by jarrodmillman** [comment]
@dschult @MridulS @rossbar I was about to just do all this in the release, but then realized it is pretty different than what was on main or in the RC. So I decided to make it a PR to make the history easier to follow. It would be great if someone could approve it, but I will probably just self-merge it in a bit since I already shared the script I used earlier with you on another channel.

File: doc/release/release_dev.rst [hunk 1]
```diff
@@ -208,53 +227,94 @@ A total of 217 changes have been committed.
 - PR for issue #6033 Improve test coverage for algorithms in betweenness_subset.py #6033 (#6083)
 - Di graph edges doc fix (#6108)
 - Improve coverage for core.py (#6116)
-- Add clear_edges() method to the list of methods to be frozen by nx.freeze() (#6190)
+- Add clear edges method to the list of methods to be frozen by the nx.… (#6190)
```
Comment (rossbar) [inline doc/release/release_dev.rst:230]: The formatting of this one looks a little weird with the unicode ellipsis, but no big deal.

**Review by rossbar — APPROVED** [review]
Thanks @jarrodmillman , it's no small task putting these together! I looked through everything and thought most things looked generally correct, both in terms of which PRs are included and the authors.

File: doc/release/release_dev.rst (hunk 1 above)
Comment (dschult) [reply doc/release/release_dev.rst:230]: I changed the title of the #6190 PR to be `Add clear edges method as a method to be frozen by nx.freeze`.  So:
```suggestion
- Add clear edges method as a method to be frozen by nx.freeze (#6190)
```
This should be what the script produces now.

File: doc/release/release_dev.rst [hunk 2]
```diff
@@ -208,53 +227,94 @@ A total of 217 changes have been committed.
...
+- [DOC] Follow numpydoc standard in barbell_graph documentation (#6286)
+- Update simple_paths.py: consistent behaviour for `is_simple_path` when path contains nodes not in the graph. (#6272)
+- Correctly point towards 2.8.8 in release notes (#6298)
+- Isomorphism improve documentation (#6295)
+- Improvements and test coverage for `line.py` (#6215)
+- Fix typo in Katz centrality comment (#6310)
+- Broken link in isomorphism documentation (#6296)
+- Update copyright years to 2023 (#6322)
+- fix warnings for make doctest (#6323)
+- fix whitespace issue in 'networkx.generators.tests.test_internet_as_g… (#6324)
```
Comment (dschult) [inline doc/release/release_dev.rst:267]: This is a similar ellipsis issue, but the title of the PR had already been changed... unfortunately after the commit was merged...  So I guess changing the title of the PR after merging does not affect the git log. And that makes sense...  But it means that the script will not produce either of the PR titles I'm suggesting we use in the release notes.
```suggestion
- fix whitespace issue in test_internet_as_graph (#6324)
```

**Review by dschult — COMMENTED** [review]
I made two suggestions to revise the PR titles for the two cases of ellipsis (when the original title was too long). I'm fine with however these are treated.

We should also consider mentioning the backend infrastructure (#6000) in the Highlights section as well as drawing using LatTeX/TikZ (#6238), and VF2++ isomorphism algorithm (#5788).  I'm not sure if this PR extends to the "Highlights" section, or only the output of the script.

**Comment by jarrodmillman** [comment]
@dschult I added something about LatTeX/TikZ (https://github.com/networkx/networkx/pull/6238), and the VF2++ isomorphism algorithm (https://github.com/networkx/networkx/pull/5788) to the highlights section. Mridul already added something about the backends, but I added a link to the PR (since it may be of particular interest to users).

Feel free to update this PR and merge when the tests pass. I am going to drive to San jose in a bit and will make the release from there this afternoon. Now that the release notes are fixed, the release will be super easy. Thanks!!

**Review by dschult — APPROVED** [review]
Excellent!  Thanks for those changes @jarrodmillman And for knowing that @MridulS had already added text about the backend stuff.  I'm going to assume that Mridul has added any desired highlights already and merge this so it's ready for a release.

**Pull request merged**